docs/tutorials/annotation/annotate_points.md

```diff
@@ -72,10 +72,14 @@ def create_label_menu(points_layer, labels):
 
     def label_changed(event):
         """Update the Points layer when the label menu selection changes"""
-        selected_label = event.value
+        try:
+            selected_label = event.value
+        except AttributeError:
+            selected_label = event
         current_properties = points_layer.current_properties
         current_properties['label'] = np.asarray([selected_label])
         points_layer.current_properties = current_properties
+        points_layer.refresh_colors()
```

Comment:
Without this the point colors do not get updated on label change.

Reply:
I *think* `current_properties` is deprecated and you should use `feature_defaults` instead. @andy-sweet?

Reply:
I have now resolved the other issues, but simply replacing `current_properties` by `feature_defaults` everywhere does no fully work. The viewer starts, but changing the label (via menu or shortcut) does not change it for the currently selected point. It however changes for the next point that is placed.

I made two videos to demonstrate: first is the version with `current_properties` where everything works as expected (I am using a shortcut to flip the label in the video, but I also checked that changing it via the menu works):

https://user-images.githubusercontent.com/4263537/235505773-8242a255-d7fe-48aa-a89d-79de0bfb605e.mp4

And this is the version where I have replaced all `current_properties` with `feature_defaults`. Now switching the label for the currently selected point doesn't work anymore, but the next point that is placed has the correct label.

https://user-images.githubusercontent.com/4263537/235507388-6c13052c-a8ac-4fca-a7ad-c0a98b304a4b.mp4

I am not sure if I am doing something wrong or if there is still a bug.

cc @andy-sweet

Reply:
`current_properties` and `feature_defaults` are not equivalent; this is unfortunately a conflation we have with points that [we're currently trying to resolve](https://github.com/napari/napari/pull/5722#discussion_r1165758964).

In this case, this means that `current_properties` affects both the currently selected point, and the default values for the following point. However (due to a different issue which is [solved with `feature_defaults`](https://github.com/napari/napari/pull/5646)) it does not correctly update everything that needs to be updated.

On the other hand, `feature_defaults` *does* work as intended, but it intentionally does not affect the `current` features.

In short, I *think* you should probably change *both* here ^^' but maybe @andy-sweet knows better.

Reply:
@brisvag's summary is correct. They are intentionally not equivalent in an effort to simplify what `feature_defaults` means and what its setter does, compared to `current_properties`. I understand that's annoying when you need the extra functionality of `current_properties` - maybe we can just add a separate method to update the currently selected feature values (e.g. `Points.update_selected_features(...)`) to help real usage and keep things clear? Note that some similar discussion came up in https://github.com/napari/napari/pull/4218 where we updated a similar, simpler example.

Also, technically, `current_properties` is not deprecated yet (i.e. there is no deprecation warning). There is an intent to deprecate `properties`, `current_properties`, and `property_choices` (see https://github.com/napari/napari/pull/5477), but that intent has existed for some time, so I don't want to make predictions about when it will actually happen...

My instinct is to use `current_properties` for now since the whole example refers to the properties family and either workaround any bugs (i.e. by using `refresh_colors`) or file a bug related to the `current_properties` setter, so that can be fixed. Then the properties -> features swap should occur as part of the whole deprecation by whoever takes that me (i.e. probably me?).

And thanks for the effort here! If you're interested in trying to work through the nasty details here, I'm happy to help to make features work, but I also don't want people to miss out on a useful update to the docs.

Reply:
Thanks for the explanations @brisvag and @andy-sweet. I would then suggest to move ahead and keep using `current_properties` for now; and we can revisit this once you actually deprecate it.
What do you think @jni? If you agree I can finish this up since the changes make the example run again.